cmake/SetupCompilerOptions.cmake

```diff
@@ -261,13 +265,23 @@ elseif( BLT_CXX_STD STREQUAL c++11 )
         FLAGS_VAR CMAKE_CXX_FLAGS
         DEFAULT "-std=c++11")
     endif ()
+    if (COMPILER_FAMILY_IS_PGI)
```

Comment:
Thanks @kennyweiss, I am still checking on the CMAKE_CXX_STANDARD question...  Do you want to hold your approval until after I finish?

Reply:
As suggested, I ran with ``` -DCMAKE_EXPORT_COMPILE_COMMANDS=1 ``` and noted that cmake does not appear to be adding the appropriate options for c++11 for PGI:

``` C++
  "command": "/usr/tce/packages/pgi/pgi-17.10/bin/pgc++  -DGTEST_HAS_DEATH_TEST=1 -I/g/g0/martymcf/umpire/build/linux-rhel7-x86_64/umpire/include -I/g/g0/martymcf/umpire/src/umpire/blt/thirdparty_builtin/googletest-master-2018-02-01/googlemock/include -I/g/g0/martymcf/umpire/src/umpire/blt/thirdparty_builtin/googletest-master-2018-02-01/googlemock -I/g/g0/martymcf/umpire/src/umpire/blt/thirdparty_builtin/googletest-master-2018-02-01/googletest/include -I/g/g0/martymcf/umpire/src/umpire/blt/thirdparty_builtin/googletest-master-2018-02-01/googletest    -Minform=warn  -fPIC     -Minform=warn   -DGTEST_HAS_PTHREAD=1   -o CMakeFiles/gmock_main.dir/src/gmock_main.cc.o -c /g/g0/martymcf/umpire/src/umpire/blt/thirdparty_builtin/googletest-master-2018-02-01/googlemock/src/gmock_main.cc",
```

Reply:
So I think the new logic may need to stay.

Reply:
Thanks for checking, and adding the logic!

Since there is more than one case now, we can modify the logic to:
```cmake
    ...
elseif( BLT_CXX_STD STREQUAL c++11 )
    set(CMAKE_CXX_STANDARD 11)
    blt_append_custom_compiler_flag(
        FLAGS_VAR CMAKE_CXX_FLAGS
        DEFAULT " "
        XL "-std=c++11"
        PGI "--c++11")
elseif( BLT_CXX_STD STREQUAL c++14)
    set(CMAKE_CXX_STANDARD 14)
    blt_append_custom_compiler_flag(
        FLAGS_VAR CMAKE_CXX_FLAGS
        DEFAULT " "
        XL "-std=c++1y"
        PGI "--c++14")
endif ()
```

which is cleaner and more easily extensible.